supertokens_python/ingredients/emaildelivery/services/smtp.py

```diff
@@ -0,0 +1,121 @@
+# obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+
+import smtplib
+import ssl
+from abc import ABC, abstractmethod
+from email.mime.text import MIMEText
+from typing import Any, Callable, Dict, Generic, TypeVar, Union
+
+from supertokens_python.logger import log_debug_message
+
+_T = TypeVar('_T')
+
+
+class SMTPServiceConfigFrom:
+    def __init__(self, name: str, email: str) -> None:
+        self.name = name
+        self.email = email
+
+
+class SMTPServiceConfig:
+    def __init__(
+        self, host: str,
+        port: int,
+        from_: SMTPServiceConfigFrom,
+        password: Union[str, None] = None,
+        secure: Union[bool, None] = None,
+    ) -> None:
+        self.host = host
+        self.from_ = from_
+        self.password = password
+        self.port = port
+        self.secure = secure
+
+
+class GetContentResult:
+    def __init__(self, body: str, subject: str, to_email: str, is_html: bool = False) -> None:
+        self.body = body
+        self.subject = subject
+        self.to_email = to_email
+        self.is_html = is_html
+
+
+class Transporter:
+    def __init__(self, smtp_settings: SMTPServiceConfig) -> None:
+        self.smtp_settings = smtp_settings
+
+    def _connect(self):
+        try:
+            mail = smtplib.SMTP(self.smtp_settings.host, self.smtp_settings.port)
+            if self.smtp_settings.secure:
+                context = ssl.create_default_context()
+                mail.starttls(context=context)
+
+            if self.smtp_settings.password:
+                mail.login(self.smtp_settings.from_.email, self.smtp_settings.password)
+
+            return mail
+        except Exception as e:
+            log_debug_message("Couldn't connect to the SMTP server: %s", e)
+            raise e
+
+    async def send_email(self, input_: GetContentResult,
+                         _: Dict[str, Any]) -> None:
+        connection = self._connect()
+        if connection is None:
+            raise Exception("Couldn't connect to the SMTP server.")
+
+        from_ = self.smtp_settings.from_
+        try:
+            from_addr = f"{from_.name} <{from_.email}>"
+            if input_.is_html:
+                email_content = MIMEText(input_.body, "html")
+                email_content["From"] = from_addr
+                email_content["To"] = input_.to_email
+                email_content["Subject"] = input_.subject
+                connection.sendmail(from_.email, input_.to_email, email_content.as_string())
+            else:
+                connection.sendmail(from_addr, input_.to_email, input_.body)
```

Comment:
Yes. This is what the in-built `smtplib` provides. It isn't a async function.

Reply:
Are you sure there is no async version of this? Cause if this is blocking, then it will block the webserver from serving any requests while this is running..